src/main/webapp/scripts/map.js

```diff
@@ -12,82 +12,145 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/* Global Icon Variables */
+var yellowIcon;
+var blueIcon;
+var purpleIcon;
+var redIcon;
+var greyIcon;
 
+/* Hard-coded Recommendations */
+const recommendations = [
+  ['Places to Visit', 'Sky Garden', 51.510881, -0.083751, 'Sky Garden is a great place to visit, with really good views of London. \
+  You can enjoy the 360 degree view of the city - for free! They are public gardens and definitely worth a visit \
+  when you have the time. Remember to keep socially distanced!'],
+  ['Restaurants', 'Mildreds Kings Cross', 51.531299, -0.11716, 'A lovely little place to grab something to eat! There are a lot of \
+  different options and a great atmosphere inside Mildred\'s!'],
+  ['Bars and Clubs', 'Fitz\'s Bar', 51.517735, -0.097112, 'This is a recommendation for Fitz\'s Bar in London.'],
+  ['Study Places', 'Bloomsbury Coffee House', 51.525212, -0.126469, 'A really great place to study!']
+]
 
-/* Custom Markers */
-const yellowIcon = {
-  url: "/images/yellow-marker.png",
-  scaledSize: new google.maps.Size(30, 40),
-};
-
-const blueIcon = {
-  url: "/images/blue-marker.png",
-  scaledSize: new google.maps.Size(30, 40),
-};
-
-const purpleIcon = {
-  url: "/images/purple-marker.png", 
-  scaledSize: new google.maps.Size(30, 40),
-};
-
-const redIcon = {
-  url: "/images/red-marker.png", 
-  scaledSize: new google.maps.Size(30, 40),
-};
+window.onload = function() {
+  /* Custom Markers */
+  yellowIcon = {
+    url: "/images/yellow-marker.png",
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  blueIcon = {
+    url: "/images/blue-marker.png",
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  purpleIcon = {
+    url: "/images/purple-marker.png", 
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  redIcon = {
+    url: "/images/red-marker.png", 
+    scaledSize: new google.maps.Size(30, 40)
+  };
+  greyIcon = {
+    url: "/images/grey-marker.png",
+    scaledSize: new google.maps.Size(30, 40)
+  }
+  initMap();
+}
```

Comment:
Please check the comment I added in `recommendation-map.html` We can simply git rid of this function by having the markers as constants defined in the global scope